.github/workflows/clang-tidy.yaml

```diff
@@ -0,0 +1,26 @@
+name: clang-tidy
+
+on: [pull_request]
+
+jobs:
+  job:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Install GoogleTest.
+        run: sudo apt install -y libgtest-dev
+
+      - name: Generate build files and compilation database.
+        run: cmake -B out
+
+      - name: Lint with clang-tidy.
+        uses: ZedThree/clang-tidy-review@v0.7.1
```

Comment:
We may need to change the version `v0.7.1`, because we want to use the clang-tidy-12 which is supported in a very recent commit of clang-tidy-review.